### ISSUE TITLE
Another @everyone @here vulnerability

### ISSUE THREAD

**Issue by TS-7** [utarwyn/discord-tictactoe#20]
Hey there! Your module is really amazing and I do not want to replace it with any other ones. At the moment I have observed an @everyone vulnerability where if the person's nickname is @everyone ... It is simply going to ping @ everyone ... 

I made small changes to your package on my fork - but I am unable to build it on my discord bot for some reason. Using my discord fork link in the package.json file gives plenty errors. 
If you could help me with either by fixing the vulnerability or buy getting my fork to work - that'd be really really appreciated.

Thanks!

**Comment by utarwyn** [comment]
Hello! Thank you for your feedback. Where can I find your fork? I'm interested about your changes because other server owners also get issues with @everyone and @here.

**Comment by TS-7** [comment]
Thanks for the quick response! You can find my fork here : https://github.com/IYR3775/discord-tictactoe
The only changes are in the en.json locale file. My team has added a small edit to avoid the everyone ping and have made custom messages for the same. Please let me know if anything.

Thanks!

**Comment by utarwyn** [comment]
Thank you for the link. I have checked the repository but did not found any commit related to the tagging system? Have you an idea of a solution to avoid tagging @everyone and @here?

**Comment by TS-7** [comment]
Thanks for checking it out! I guess you did not observe the json ... In the json game object - I've changed the title to have ``` before and after. This is a discord mark down to convert text into codeblocks. With how Discord works - codeblocks don't allow you to ping a user nor @everyone @here. That's how I fixed it on my repo. And also have improvised the English text and made it more fancy with nice emojis. If you want to test it - let me know so I can open a pull request 🙂

**Comment by utarwyn** [comment]
Oh ok sorry.. I have seen your changes in the language file, very interesting. But it breaks the style of some messages, because I would like to also apply the rule for request messages.

The main issue is that I am using field `displayName` to get the display name of a guild member and it can contains special characters (like `@`). A lot of other bots use field `user.username` to bypass the problem but they do not know if a member modifies its name. I'm still searching for a working fix (maybe by removing all special characters from displayname, I don't know..). 

I have already worked on a similar issue #13.

**Comment by TS-7** [comment]
Thanks for investing time into the issue ... At the moment - I most likely will have the bot disable @ everyone mentions since we don't use it anywhere in the handler! But here's our public bot. https://discord.com/oauth2/authorize?client_id=723112579584491571&permissions=2147483135&scope=bot%20applications.commands I use the same fork I gave you previously - seems like everything is flawless - feel free to take a look! It's currently in 75 servers. Just a couple steps until verification but we might run into problems when they identify the ping vulnerability so just want to enquire. 

Thank you so much for your quick attention! I do appreciate it. Have a great night

**Comment by utarwyn** [comment]
Indeed your solution is very nice. I have added quotes around member display names to avoid unwanted tagging. I took the opportunity to add some emojis like you have done in your fork 😉 

Hope this commit will resolve all issues with member tagging.
Can you test it? If everything is ok for you, I can tag a new version!

Thank you for your detailed feedback, very appreciated.

**Comment by TS-7** [comment]
Sorry for the delayed response. Got caught up with work. Your latest push fixes all the vulnerability I have observed and the emoji's are a wonderful addition! Thank you so much for being so quick. I will go ahead and recommend this module to all my friendly devs! Thanks a lot again utarwyn. The module at this stage is perfect. Much much appreciated mate

**Comment by TS-7** [comment]
I have noticed another thing which probably should go in another issue. Let me know if I should open a new one but everytime I run $ttt for the first time - it doesn't do anything. It happens to all servers - the first usage is always not working. If it is initialising or something similar, could we have a message say the command is being initialised ? I can send a small clip of what happens when the bot restarts and ttt is ran for the first time. But after that it is flawlessly incredible!

**Comment by utarwyn** [comment]
Thank you so much! I have never seen that initialisation problem before. Are you sure that the tictactoe bot is created during your bot startup? If you have a small clip about it, I am interested!

**Comment by TS-7** [comment]
Sorry, couldn't stay in touch. I've prepared the small clip of the issue. Check it out and let me know. 

Thanks
https://user-images.githubusercontent.com/63742049/105965269-359aa980-60a9-11eb-8e63-e87ae91c4d1a.mp4

**Comment by utarwyn** [comment]
I understand the issue I think: you are using a custom command handling system.
You only start the TicTacToe module when your custom command is typed (`async run` method), with this part of your code:
```js
new TicTacToe({
  language: 'en',
  command: `${data.guild.prefix}ttt`
}, client);
```

After this code is executed, my module will handle tictactoe commands. This explain why you have to type command twice when restarting your bot: first time for your custom command, others handled by my module. So two possible options as my point of view:

1. You move this part of code which starts TicTacToe module at your bot startup ;
2. **OR** I have to add a feature that enables usage of custom commands system.

I hope my explanation is clear enough.. Tell me which option do you prefer.

**Comment by TS-7** [comment]
Absolutely! Thanks for the clarification - that was very transparent. So with how my team has organised the project - we haven't handled commands within the bot startup file. It would mess the functions up too is what I feel. I have another sub project that I will use your wonderful module in and I would vote for the second option. It will personally help you too in the long run because I am confident there's many others who want to use the command within their custom handler because none of the bot developers use a single file to handle their commands. 
I would really appreciate you helping us out with that. 

Thanks again for your amazing response times! I can't thank you enough.

**Comment by utarwyn** [comment]
Ok I will work on that! I keep you informed, maybe we can discuss that on another issue 👍

**Comment by johan-perso** [comment]
> Thank you for the link. I have checked the repository but did not found any commit related to the tagging system? Have you an idea of a solution to avoid tagging @everyone and @here?

I think if the username / nickname include a "@", the bot should return a message and stop the game. This can be good and fix that.

**Comment by TS-7** [comment]
Hey utarwyn! Good to talk to you again. Is there any updates / progress on the module ? Just checking in :)

**Comment by utarwyn** [comment]
Hello! Yes this evening :+1:

**Comment by utarwyn** [comment]
I have just released 2.0.8 to fix the issue! :smile: 
https://github.com/utarwyn/discord-tictactoe/releases/tag/v2.0.8
